File: slim_stat.pl



my $count;
my $sum;
my $first;
my $last;

open(BLOC, "$ARGV[0]") or die $!;
while(<BLOC>) {
	        my $line = <BLOC>;
	        $line = sprintf("%.3f", $line);
	        $sum = $sum + $line;
	        print $line,"\n";
	        print $sum,"\n";
	        $count = $count + 1;
	    }
close(BLOC);
print $sum,"\n";
print $count,"\n";

my $mean = $sum / $count;
my $lastCount = $count;
my $total_dev;
my $count;

open(BLOC, "$ARGV[0]") or die $!;
while(<BLOC>) {
	        my $line = <BLOC>;
	        $line = sprintf("%.3f", $line);
	       	if ($count == 0){
	        	$first = $line;
	        }
	        elsif($count == $lastCount){
	        	$last = $line;
	        }
	        else{}
	        my $dev = ( $line - $mean )**2;
	        $total_dev = $total_dev + $dev;
	        $count = $count + 1;

	    }
close(BLOC);
my $sd =  sqrt($total_dev / ($count - 1));
my $se = $sd / sqrt($count);
my $slope = ($last - $first) / $count;

print $mean, " ", $se, " ", $slope, "\n";

######

File: slim_stat.py
# -*- coding: utf-8 -*-
#python
#Felix Beaudry Aug 10 2018

import fileinput, argparse
import sys
import numpy
import math

def arguments():
        parser  = argparse.ArgumentParser(description="script for script slim scripts")
        parser.add_argument("-i","--input",help="input file",required=True)
        args = parser.parse_args()
        return(args)

args = arguments()
infile = args.input

with open(infile) as f:
    content = f.readlines()
content = [line.strip() for line in content] 
floaties = [float(i) for i in content]
sumOf = sum(floaties)
mean = sumOf / len(content)
	
slope = (floaties[len(content) -1 ] - floaties[0]) / len(content)

#se
totalDev = 0
for x in range(0,(len(content) -1)):
	dev = (floaties[x] - mean)**2
	totalDev = totalDev + dev
sd = math.sqrt(totalDev / (len(content) -1))
se = sd / math.sqrt((len(content) -1))
print mean, " ", se ," ", slope



